test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/UserInterfaceService/NuGetUITests.cs

```diff
@@ -144,48 +147,158 @@ public void ShowError_WhenArgumentIsRemoteInvocationExceptionForOtherException_S
             return CreateNuGetUI(Mock.Of<INuGetUILogger>(), Mock.Of<INuGetUILogger>());
         }
-
         private NuGetUI CreateNuGetUI(INuGetUILogger defaultLogger, INuGetUILogger projectLogger)
         {
-            var uiContext = CreateNuGetUIContext();
+            return CreateNuGetUI(defaultLogger, projectLogger, activeFilter: ItemFilter.All, isSolution: false, mockPackageSourceMapping: null);
+        }
+
+        private NuGetUI CreateNuGetUI(INuGetUILogger defaultLogger, INuGetUILogger projectLogger, ItemFilter activeFilter, bool isSolution, Mock<PackageSourceMapping> mockPackageSourceMapping)
+        {
+            var mockNuGetUIContext = new Mock<INuGetUIContext>();
+
+            // Without this Setup, a MessageBox will be shown which could block tests indefinitely.
+            mockNuGetUIContext.Setup(_ => _.OptionsPageActivator).Returns(Mock.Of<IOptionsPageActivator>());
 
-            return new NuGetUI(
+            if (mockPackageSourceMapping != null)
+            {
+                mockNuGetUIContext.Setup(_ => _.PackageSourceMapping).Returns(mockPackageSourceMapping.Object);
+            }
+
+            var mockIPackageManagerControlViewModel = new Mock<IPackageManagerControlViewModel>();
+            mockIPackageManagerControlViewModel.SetupGet(_ => _.ActiveFilter).Returns(activeFilter);
+            mockIPackageManagerControlViewModel.SetupGet(_ => _.IsSolution).Returns(isSolution);
+
+            var nugetUI = new NuGetUI(
                 Mock.Of<ICommonOperations>(),
                 new NuGetUIProjectContext(
                     Mock.Of<ICommonOperations>(),
                     projectLogger,
                     Mock.Of<ISourceControlManagerProvider>()),
                 defaultLogger,
-                uiContext);
+                mockNuGetUIContext.Object,
+                mockIPackageManagerControlViewModel.Object);
+
+            return nugetUI;
         }
 
-        private NuGetUIContext CreateNuGetUIContext()
+        private Mock<ITelemetrySession> SetupTelemetryListener()
         {
-            var sourceRepositoryProvider = Mock.Of<ISourceRepositoryProvider>();
-            var packageManager = new NuGetPackageManager(
-                sourceRepositoryProvider,
-                Mock.Of<ISettings>(),
-                _testDirectory.Path);
-
-            return new NuGetUIContext(
-                Mock.Of<IServiceBroker>(),
-                Mock.Of<IReconnectingNuGetSearchService>(),
-                Mock.Of<IVsSolutionManager>(),
-                new NuGetSolutionManagerServiceWrapper(),
-                packageManager,
-                new UIActionEngine(
-                    sourceRepositoryProvider,
-                    packageManager,
-                    Mock.Of<INuGetLockService>()),
-                Mock.Of<IPackageRestoreManager>(),
-                Mock.Of<IOptionsPageActivator>(),
-                Mock.Of<IUserSettingsManager>(),
-                new NuGetSourcesServiceWrapper(),
```

Comment:
Most of this wasn't being used, so I reduced it and merged into `CreateNuGetUI()`